src/Microsoft.SymbolStore/KeyGenerators/ELFFileKeyGenerator.cs

```diff
@@ -57,6 +57,15 @@ public override IEnumerable<SymbolStoreKey> GetKeys(KeyTypeFlags flags)
             if (IsValid())
             {
                 byte[] buildId = _elfFile.BuildID;
+                if (buildId != null && buildId.Length < 20 && buildId.Length >= 16)
```

Comment:
I think this should be in the below static GetKeys() functions and remove the length check here.  That allows the build id to be normalized in all cases. The static GetKeys() is used by some of tools (dotnet-dump) directly with the build id which also could be the 16 byte/md5 version.

Reply:
Ah, good call. I think we need to do this in both places then because of the calls to `BuildKey` later on in this function.